pilot/pkg/networking/core/v1alpha3/route/route.go

```diff
@@ -539,6 +545,28 @@ func translateRoute(node *model.Proxy, in *networking.HTTPRoute,
 	return out
 }
 
+// splitRedirectAuthority takes an authority redirect string and translates it into a scheme, host, and port.
+// This allows redirecting any combination of the three parts, rather than the whole thing.
+func splitRedirectAuthority(redirect string) (scheme string, host string, port uint32) {
+	// url.Parse is too strict here, we need to do our own parsing
+	for _, match := range []string{"https://", "http://"} {
+		if strings.HasPrefix(redirect, match) {
+			scheme = match[:len(match)-3]
+			redirect = strings.TrimPrefix(redirect, match)
+			break
+		}
+	}
+	h, p, err := net.SplitHostPort(redirect)
+	if err != nil {
+		host = redirect
```

Comment:
I don't really understand this part...host is set to the redirect URL if the redirect is malformed?

Reply:
If there is no port -> the full thing is the host. If there is a port, host=h, port=p. I can expand the comments